app/scripts/modules/core/src/entityTag/alerts/notificationCategories.ts

```diff
@@ -0,0 +1,59 @@
+import { Dictionary, keyBy } from 'lodash';
+
+export interface INotificationCategory {
+  [key: string]: string|number;
+  id: string;
+  severity: number;
+  label: string;
+  icon: string;
+}
+
+const NOTIFICATION_CATEGORIES = [
+  {
+    id: 'srsBzns',
+    label: 'Security Vulnerability',
+    icon: 'fa-lock',
+    severity: 2,
+  },
+
+  {
+    id: 'blacklist',
+    label: 'Blacklisted',
+    icon: 'fa-ban',
+    severity: 2,
+  },
+
+  {
+    id: 'deprecation',
+    label: 'Deprecations',
+    icon: 'fa-wrench',
+    severity: 1,
+  },
+
+  {
+    id: 'default',
+    label: 'Other',
+    icon: 'fa-exclamation-circle',
+    severity: 0,
+  }
+];
+
+const BY_NAME: Dictionary<INotificationCategory> = keyBy(NOTIFICATION_CATEGORIES, 'id');
```

Comment:
it's a `lodash` specific construct and is what is returned from `keyBy()`

It's basically a shortcut for typesafe hash:
```

    interface Dictionary<T> {
        [index: string]: T;
    }
```